clientevent.go

```diff
@@ -15,39 +15,43 @@
 
 package gomatrixserverlib
 
-import (
-	"encoding/json"
-)
-
 // ClientEvent is an event which is fit for consumption by clients, in accordance with the specification.
 type ClientEvent struct {
-	Content        json.RawMessage `json:"content"`
-	Sender         string          `json:"sender"`
-	Type           string          `json:"type"`
-	StateKey       *string         `json:"state_key,omitempty"`
-	Unsigned       json.RawMessage `json:"unsigned,omitempty"`
-	OriginServerTS int64           `json:"origin_server_ts"`
-	EventID        string          `json:"event_id"`
+	Content        rawJSON `json:"content"`
+	EventID        string  `json:"event_id"`
+	OriginServerTS int64   `json:"origin_server_ts"`
+	// RoomID is omitted on /sync responses
+	RoomID   string  `json:"room_id,omitempty"`
+	Sender   string  `json:"sender"`
+	StateKey *string `json:"state_key,omitempty"`
+	Type     string  `json:"type"`
+	Unsigned rawJSON `json:"unsigned,omitempty"`
 }
 
-// ToClientEvents converts server events to client events
-func ToClientEvents(serverEvs []Event) []ClientEvent {
+// ToClientEvents converts server events to client events. If omitRoomIDs is true, the room_id will not be
+// set on the resulting events.
+func ToClientEvents(serverEvs []Event, omitRoomIDs bool) []ClientEvent {
```

Comment:
Not sure I like the omitRoomIDs being a boolean. It might be nicer to make it an enum.